File: src/main/java/com/tom/student/Student.java
package com.tom.student;

public class Student {
    String id;
    String name;
    int english;
    int math;
    static int pass = 60;

    public Student(String name, int english, int math) {
        this.name = name;
        this.english = english;
        this.math = math;
    }

    public  int highest(){
      /*  int max = 0;
        if (english > math){
            max = english;
        }else {
            max=math;
        }*/

        return  (english > math) ? english : math;
    }

    public void print(){
        int average = getAverage();
        System.out.print(name + "\t" + english + "\t" + math+
                "\t" + getAverage() + "\t" +
                ((getAverage() >= pass) ? "Pass" : "Failed"));

        char grading = 'F';
        switch (average/10){
            case 10:
            case 9:
                grading = 'A';
                break;
            case 8:
                grading = 'B';
                break;
            case 7:
                grading = 'C';
                break;
            case 6:
                grading = 'D';
                break;
            default:
                grading = 'F';
                break;
        }
        System.out.println("\t" + grading);

        /*if (average >= 90 && average <= 100){
            grading = 'A';
        }else if (average >= 80 && average <= 89){
            grading = 'B';
        }else if (average >= 70 && average <= 79){
            grading = 'C';
        }else if (average >= 60 && average <= 69){
            grading = 'D';
        }else {
            grading = 'F';
        }*/


        /*if (getAverage() >= 60){
            System.out.println("\tPass");
        }else {
            System.out.println("\tFailed");
        }*/
    }
    public int getAverage(){
        return (english+math)/2 ;
    }
}

File: src/main/kotlin/com/tom/student/Student.kt
package com.tom.kotlin

import java.util.*

fun main() {
//    userInput()
    Student.pass = 50
    val stu = Student("Eric", 77, 99)
    val stu2 = Student("Jane", 60, 40)
    val stu3 = Student("Hank", 50, 30)
    val gstu = GraduateStudent("Jack", 55,65,60)
    gstu.print()
    stu.print()
    stu2.print()
    stu3.print()
    println("Highest score: ${stu.highest()}")
}

class GraduateStudent(name: String?,english: Int,math: Int,var thesis: Int) : Student(name,english,math){
    companion object{
        var pass =70
    }

    override fun print() {
        println("$name\t$english\t$math\t$thesis\t${getAverage()}\t${passOrFailed()}\t${grading()}")
    }

    override fun passOrFailed() = if(getAverage() >= pass) "Pass" else "Failed"
}

open class Student(var name : String?,var english :Int,var math : Int ) {
    companion object{
        @JvmStatic
        var pass = 60
    }
    open fun print(){
        println("$name\t$english\t$math\t${getAverage()}\t${passOrFailed()}\t${grading()}")
    }

    open fun passOrFailed() = if(getAverage() >= pass) "Pass" else "Failed"

    fun grading() = when(getAverage()){
        in 90..100 -> 'A'
        in 80..89 -> 'B'
        in 70..79 -> 'C'
        in 60..69 -> 'D'
        else -> 'F'
    }

    fun getAverage() = (english+ math)/2
    fun highest() = if(english > math)english else math

    fun nameCheck() = name?.length
}

private fun userInput() {
    val scanner = Scanner(System.`in`)
    print("Please enter student's name: ")
    //var name = scanner.next()
    var name = null
    print("Please enter student's english: ")
    var english = scanner.nextInt()

    print("Please enter student's math: ")
    var math = scanner.nextInt()
    val stu = Student(name, english, math)
    stu.print()
    stu.nameCheck()
}

